### PR TITLE
I18N - Don't generate SchemaStructure, read schema data on-the-fly

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Part of https://lab.civicrm.org/dev/core/-/issues/4999
Followup to #29472 which converted the generated DAOs, this converts the generated `CRM_Core_I18n_SchemaStructure` class to a non-generated class that reads from the schema.

Before
----------------------------------------
`CRM_Core_I18n_SchemaStructure` consists of functions that return schema metadata, generated from the xml by GenCode.

After
----------------------------------------

`CRM_Core_I18n_SchemaStructure` no longer generated & no longer contains any data, now just reads it from the schema.

### CODE REVIEW

File: tests/phpunit/CRM/Core/I18n/SchemaTest.php [hunk 1]
```diff
@@ -112,7 +112,7 @@ public function testSchemaBuild(): void {
     $inUseCollation = CRM_Core_BAO_SchemaHandler::getInUseCollation();
     $testCreateTable = CRM_Core_DAO::executeQuery('SHOW create table civicrm_price_set', [], TRUE, NULL, FALSE, FALSE);
     while ($testCreateTable->fetch()) {
-      $this->assertStringContainsString("`title_en_US` varchar(255) COLLATE {$inUseCollation} NOT NULL DEFAULT '' COMMENT 'Displayed title for the Price Set.'", $testCreateTable->Create_Table);
+      $this->assertStringContainsString("`title_en_US` varchar(255) COLLATE {$inUseCollation} NOT NULL", $testCreateTable->Create_Table);
```
Comment (demeritcowboy) [inline tests/phpunit/CRM/Core/I18n/SchemaTest.php:115]: I'm still not sure if https://github.com/civicrm/civicrm-core/pull/27226 was the right approach, but assuming it is, then the `DEFAULT ''` is important.

Comment (colemanw) [reply tests/phpunit/CRM/Core/I18n/SchemaTest.php]: Thanks @demeritcowboy I've pushed a 2nd commit to restore that funkyness

**Comment by demeritcowboy** [comment]
The schema before and after looks good. But clicking around when I try to switching to another language it blows up. Happens on the PR test site here too:

```
Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 262144 bytes) in /home/jenkins/bknix-dfl/build/core-30310-3ymef/web/sites/all/modules/civicrm/packages/DB/DataObject.php on line 1847

Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 262144 bytes) in /home/jenkins/bknix-dfl/build/core-30310-3ymef/web/includes/database/database.inc on line 1873

Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 262144 bytes) in Unknown on line 0

Fatal error: Allowed memory size of 268435456 bytes exhausted (tried to allocate 262144 bytes) in Unknown on line 0
```

**Comment by colemanw** [comment]
Thanks @demeritcowboy that was a fun little rabbit hole. Ok I think I got all the rabbits out. Can you check again?

**Comment by demeritcowboy** [comment]
ts(ts(...['rabbit']))

File: mixin/lib/civimix-schema/src/SqlGenerator.php [hunk 2]
```diff
@@ -91,27 +91,7 @@ private function generateCreateTableSql(array $entity): string {
...
-      }
-      if (!empty($field['auto_increment'])) {
-        $fieldSql .= " AUTO_INCREMENT";
-      }
-      $fieldSql .= self::getDefaultSql($field);
-      if (!empty($field['description'])) {
-        $fieldSql .= " COMMENT '" . \CRM_Core_DAO::escapeString($field['description']) . "'";
-      }
-      $definition[] = $fieldSql;
+      $definition[] = "`$fieldName` " . \CRM_Utils_Schema::generateFieldSql($field);
```
[the remark this answers is not in the document]
Comment (colemanw) [reply mixin/lib/civimix-schema/src/SqlGenerator.php]: @totten ok, updated.

**Comment by monishdeb** [comment]
Jenkins test this please

**Comment by totten** [comment]
Looks like it does what it says. 😄 

In for a penny, in for a dollar.

**Pull request merged**